geometric_kernels/spaces/circle.py

```diff
@@ -49,7 +50,8 @@ def __call__(self, X: B.Numeric, **parameters) -> B.Numeric:
                 values.append(const * B.cos(freq * theta))
                 values.append(const * B.sin(freq * theta))
 
-        return B.concat(*values, axis=1)  # [N, M]
+        # Possibly clip last eigenfunction
+        return B.concat(*values, axis=1)[:, : self._num_eigenfunctions]  # [N, M]
```

Comment:
```suggestion
        return B.concat(*values, axis=1)[:, :self._num_eigenfunctions]  # [N, M]
```

Reply:
Can you explain when this would be necessary? I can imagine one always wants the full level?